client/my-sites/stats/activity-log-search/index.jsx

```diff
@@ -0,0 +1,87 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Search from 'components/search';
+import Gridicon from 'gridicons';
+
+class ActivityLogSearch extends Component {
+	static propTypes = {
+		onSearch: PropTypes.func,
+		translate: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		onSearch: noop,
+	};
+
+	filters = [
+		{
+			key: 'activity_type',
+			label: 'Activity type',
+			icon: 'types',
+		},
+		{
+			key: 'user',
+			label: 'User',
+			icon: 'user',
+		},
+		{
+			key: 'time',
+			label: 'Time',
+			icon: 'time',
+		},
+		{
+			key: 'post',
+			label: 'Post',
+			icon: 'posts',
+		},
+	];
+
+	renderFilter = filter => {
+		return (
+			<div className="activity-log-search__filter" key={ filter.key }>
+				<Gridicon
+					icon={ filter.icon || 'cog' }
+					className="activity-log-search__filter-icon"
+					size={ 18 }
+				/>
+				{ filter.label }
+			</div>
+		);
+	};
+
+	renderFilters = () => {
+		return this.filters.map( filter => this.renderFilter( filter ) );
```

Comment:
a few things:
 - we can just call `this.filters.map( this.renderFilter )` since we're only passing one parameter
 - this function is probably superfluous. generally we like to keep the JSX abstraction low, can we inline this?
 - since we're only returning the result of an expression we don't need the curlies :smile:
 - none of the filters are missing an icon - maybe we should just skip the default `cog`

(assume this code is below in `render()`)
```js
<div className="activity-log-search__filters-categories">
	{ this.filters.map( ( { icon, key, label } ) => (
		<div className="activity-log-search__filter" key={ key }>
			<Gridicon icon={ filter.icon } className="activity-log-search__filter-icon" size={ 18 } />
			{ filter.label }
		</div>
	) ) }
</div>
```